### ISSUE TITLE
Ability to walk around with WASD (in addition to arrow keys)

### ISSUE THREAD

**Issue by 0xdevalias** [sparkletown/sparkle#591]
It would be cool if we were able to walk around the paddock using WASD keys, in addition to the current arrow keys.

It looks like the simplest place to add this without needing to make major changes to other areas of the app is in https://github.com/sparkletown/sparkle/blob/staging/src/components/templates/Playa/MyAvatar.tsx#L195-L205

A naive minimal implementation might be as simple as:

```diff
    const keyListener = (event: KeyboardEvent) => {
      switch (event.key) {
+       case "w":
+         pressedKeys["ArrowUp"] = event.type === "keydown";
+         break;
+       case "a":
+         pressedKeys["ArrowLeft"] = event.type === "keydown";
+         break;
+       case "s":
+         pressedKeys["ArrowDown"] = event.type === "keydown";
+         break;
+       case "d":
+         pressedKeys["ArrowRight"] = event.type === "keydown";
+         break;
        case "ArrowLeft":
        case "ArrowRight":
        case "ArrowUp":
        case "ArrowDown":
          pressedKeys[event.key] = event.type === "keydown";
          break;
        default:
          return;
      }
```

**Comment by thecadams** [comment]
Hey, another great idea, thanks for this! Would be very happy to accept a PR. Want some help getting started with the dev environment?

**Comment by 0xdevalias** [comment]
Assuming it's not already documented somewhere, yeah, that would be cool :) and can write it up for anyone else who wants to jump in in future too.

**Comment by thecadams** [comment]
OK sounds good, what's the best way to reach you?

**Comment by 0xdevalias** [comment]
Twitter, keybase, email; pretty much anywhere :p

I'll drop you an email to the one on your profile.

**Comment by thecadams** [comment]
Fantastic, sounds good!

**Comment by thecadams** [comment]
Thanks for all the suggestions - closing this one - hope you can put together a PR for it!

**Comment by 0xdevalias** [comment]
Implemented in 0b459c913ed2a78fa0fdf4e2b7b7184d8b864f53 as part of https://github.com/sparkletown/sparkle/pull/732

https://github.com/sparkletown/sparkle/blob/d884f8e7822648aff1203494845f08d29cb2d085/src/hooks/useKeyboardControls.ts#L8-L14
